packages/flame/lib/src/effects/sequence_effect.dart

```diff
@@ -0,0 +1,203 @@
+import 'controllers/effect_controller.dart';
+import 'controllers/infinite_effect_controller.dart';
+import 'controllers/repeated_effect_controller.dart';
+import 'effect.dart';
+
+/// Run multiple effects in a sequence, one after another.
+///
+/// The provided effects will be added as child components; however the custom
+/// `updateTree()` implementation ensures that only one of them runs at any
+/// point in time. The flags `paused` or `removeOnFinish` will be ignored for
+/// children effects.
+///
+/// If the `alternate` flag is provided, then the sequence will run in the
+/// reverse after it ran forward.
+///
+/// Parameter `repeatCount` will make the sequence repeat a certain number of
+/// times. If `alternate` is also true, then the sequence will first run
+/// forward, then back, and then repeat this pattern `repeatCount` times in
+/// total.
+///
+/// The flag `infinite = true` makes the sequence repeat infinitely. This is
+/// equivalent to setting `repeatCount = infinity`. If both the `infinite` and
+/// the `repeatCount` parameters are given, then `infinite` takes precedence.
```

Comment:
we should just throw in these cases, better let the user know explicitly rather than trying to assume

Reply:
ok